tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/CiamIntegrationTests.cs

```diff
@@ -39,33 +39,44 @@ public async Task ROPC_Ciam_Async(string authorityFormat, int authorityVersion)
                 PublicClient = PublicClient.no
             }).ConfigureAwait(false);
 
-            string authority = string.Empty;
+            string authority = ComputeCiamAuthority(authorityFormat, authorityVersion, labResponse);
 
-            //Compute authority from format and lab response
-            switch (authorityVersion)
+            //Acquire tokens
+            string appId;
+            string email;
+            string credential;
+
+            if (authorityVersion < 3)
```

Comment:
Can you please write a new test and use private methods to avoid code duplication? This "version" idea makes the existing test too hard to read.